src/caret_analyze/common/type_check_decorator.py

```diff
@@ -246,14 +300,36 @@ def _custom_wrapper(*args, **kwargs):
 
                 if varargs_name is not None:
                     args = args[:arg_len] + _parse_collection_or_unpack(args[arg_len:])
+                sig = inspect.signature(func)
+                bound_args = sig.bind(*args, **kwargs)
+                args_dict = bound_args.arguments
                 return validate_arguments_wrapper(*args, **kwargs)
 
             except ValidationError as e:
+
                 loc_tuple = e.errors()[0]['loc']
                 annotations = get_annotations(func)
 
+                list_key = []
+                for key_value in args_dict.keys():
+                    list_key.append(key_value)
+                if varargs_name is None:
+                    loc_tuple = _change_loc_tuple(args_dict, loc_tuple, list_key)
+                else:
+                    loc_tuple = _change_loc_tuple_varargs_name(args_dict, loc_tuple, list_key)
+
+                is_method = 'self' in args_dict.keys()
+                if is_method:
+                    args = args[1:]
                 given_arg = _get_given_arg(annotations, args, kwargs, loc_tuple, varargs_name)
+                is_method = 'self' in args_dict.keys()
```

Comment:
This line was probably forgotten to be erased. It works fine without it.
```suggestion
```

Reply:
deleted